File: 53.最大子数组和.cpp
/*
 * @lc app=leetcode.cn id=53 lang=cpp
 *
 * [53] 最大子数组和
 */

// @lc code=start
class Solution {
public:
    int maxSubArray(vector<int>& nums) {
        int maxSum = nums[0], sum = 0;
        for (int i = 0; i < nums.size(); i++) {
            sum += nums[i];
            maxSum = max(sum, maxSum);
            if (sum < 0) sum = 0;
        }
        return maxSum;
    }
};
// @lc code=end

File: 53.最大子数组和.py
#
# @lc app=leetcode.cn id=53 lang=python3
#
# [53] 最大子数组和
#

# @lc code=start
class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        sum = 0
        maxSum = nums[0]
        for n in nums:
            sum = sum + n
            maxSum = max(maxSum, sum)
            sum = max(sum, 0)
        return maxSum
# @lc code=end

